chainerx_cc/chainerx/native/native_device/pool.cc

```diff
@@ -220,7 +214,7 @@ class NativeAveragePoolForwardBackward : public chainerx::AveragePoolForwardBack
 
         switch (pad_mode_) {
             case AveragePoolPadMode::kZero:
-                Mean(col, kernel_axes, out);
+                out = Mean(col, kernel_axes);
```

Comment:
This `Mean` is a routine meaning that it has its own logic for deciding the output dtype (and connect the computational graph) which is not really what we want. Here we just want to compute the result.

Reply:
Oh, I see. I totally missed it. Thank you